src/turbolinks/history.coffee

```diff
@@ -20,16 +23,27 @@ class Turbolinks.History
 
   push: (location, restorationIdentifier) ->
     location = Turbolinks.Location.wrap(location)
+    locationStack.push(location)
     @update("push", location, restorationIdentifier)
 
   replace: (location, restorationIdentifier) ->
     location = Turbolinks.Location.wrap(location)
     @update("replace", location, restorationIdentifier)
 
+  back: (location, restorationIdentifier) ->
+    location = Turbolinks.Location.wrap(location)
+    
+    previousUrl = locationStack[locationStack.length - 2]
+    if previousUrl?.absoluteURL == location.absoluteURL
```

Comment:
`Turbolinks.Location` has a handy [`isEqualTo`](https://github.com/turbolinks/turbolinks/blob/c340c42e2ece8c057aec611074971462a6e76c71/src/turbolinks/location.coffee#L43-L44) method which handles this check, so this could be update to:
```suggestion
    if location.isEqualTo(previousUrl)
```

and perhaps `previousUrl` could be renamed to `previousLocation`?